grammar.y

```diff
@@ -4,13 +4,33 @@ package sqlparser
 import "bytes"
 
 var keywordsNotAllowed = map[string]struct{}{
-  // non-deterministic keywords
+  // We don't allow non-deterministic keywords as identifiers. 
```

Comment:
Taking this as an opportunity to clarify something about non-deterministic keywords (nothing to do with this PR). This approach for non-deterministic keywords is not entirely correct. 

We are blocking the use of `current_time` in statements like `CREATE TABLE t (current_time TEXT)`, which should be allowed. We only need to block cases where `current_time` is interpreted as a value and not a column like `SELECT current_time FROM t` or `INSERT INTO t VALUES (current_time)`. But making that differentiation is a bit more complex than not allowing it in all cases. 

If this turns out to be problematic we can change it later.

Reply:
Reminder: we can still do `CREATE TABLE t ("current_time" TEXT)`

Reply:
Cool, maybe we should track it to fix it in the future.
I'm not sure if it's worth adding this clarification in the spec since probably nobody will be reading the comments. Or we can avoid adding this to the spec, and simply wait until we fix it as a known limitation in the future.

Technically, fixing this would require a fresh sync in the validator; but these cases would probably appear to have a very low probability.